### DISCUSSION TITLE
New Material Definition

### DISCUSSION THREAD

**Discussion by manikumd22** [idaholab/moose#28533] [Q&A Modules: Solid mechanics] [answered]
Hi @GiudGiud 

I am trying to define a new material module for my project, but I am getting an error

First I defined the .C and .h in src/materials and then compile the make under the folder myApp

.C file
```
#include "ComputeMultiscaleStress_circle_less_Stiff.h"
#include <random>

registerMooseObject("MooseApp", ComputeMultiscaleStress_circle_less_Stiff);

InputParameters
ComputeMultiscaleStress_circle_less_Stiff::validParams()
{
  InputParameters params = ComputeLagrangianStressPK2::validParams();
  params.addParam<MaterialPropertyName>("lambda",
                                        "lambda",
                                        "Parameter conjugate to Lame parameter"
                                        " for small deformations");
....
```

.h file
```
#pragma once

#include "ComputeLagrangianStressPK2.h"
#include "DenseMatrix.h"

class ComputeMultiscaleStress_circle_less_Stiff : public ComputeLagrangianStressPK2
{
public:
  static InputParameters validParams();
  ComputeMultiscaleStress_circle_less_Stiff(const InputParameters & parameters);

protected:
  /// Actual stress/Jacobian update
  virtual void computeQpPK2Stress();

protected:
```

```
[31m
*** ERROR ***
A 'ComputeMultiscaleStress_circle_less_stiff' is not a registered object.

If you are trying to find this object in a dynamically loaded library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.[39m

application called MPI_Abort(MPI_COMM_WORLD, 1) - process 2
```

<img width="381" alt="Screenshot 2024-09-03 at 11 29 47 PM" src="https://github.com/user-attachments/assets/1c8c2c85-709b-413f-a3c2-dd8a6b3c444d">

**Comment by GiudGiud** [comment]
hello

The headers seem to be nested under src/materials. We expect them to be under include/materials here

That's probably not the problem.
Did the application recompile correctly? Can you use the other objects you created?

Guillaume

**Comment by manikumd22** [comment]
Hi @GiudGiud 

It compiled correctly and I am getting the same error "Not a registered object".

I have tried to put in include/materials but the same error persists.


A few months back, I defined in the same way src/materials and it was working correctly. I re-installed the moose as I was some other issues with login-id 

Now after re-installment I am facing this issue

**Reply by GiudGiud** [reply, marked as answer]
ok so: 
- save all your work with git commit
- git clean -Xfd in the root directory of the app
- recompile

if it still doesnt work, same but git cleaning moose as well before recompiling

**Comment by manikumd22** [comment]
Hi @GiudGiud 

It worked


Thank you so much for the help
